rs/ethereum/cketh/minter/src/deposit.rs

```diff
@@ -403,3 +169,186 @@ pub async fn update_last_observed_block_number() -> Option<BlockNumber> {
         }
     }
 }
+
+async fn scrape_until_block<S>(last_block_number: BlockNumber, max_block_spread: u16)
+where
+    S: LogScraping,
+{
+    let scrape = match read_state(S::next_scrape) {
+        Some(s) => s,
+        None => {
+            log!(
+                DEBUG,
+                "[scrape_contract_logs]: skipping scrapping {} logs: not active",
+                S::display_id()
+            );
+            return;
+        }
+    };
+    let block_range = BlockRangeInclusive::new(
+        scrape
+            .last_scraped_block_number
+            .checked_increment()
+            .unwrap_or(BlockNumber::MAX),
+        last_block_number,
+    );
+    log!(
+        DEBUG,
+        "[scrape_contract_logs]: Scraping {} logs in block range {block_range}",
+        S::display_id()
+    );
+    let rpc_client = read_state(EthRpcClient::from_state);
+    for block_range in block_range.into_chunks(max_block_spread) {
+        match scrape_block_range::<S>(
+            &rpc_client,
+            scrape.contract_address,
+            scrape.topics.clone(),
+            block_range.clone(),
+        )
+        .await
+        {
+            Ok(()) => {}
+            Err(e) => {
+                log!(
+                    INFO,
+                    "[scrape_contract_logs]: Failed to scrape {} logs in range {block_range}: {e:?}",
+                    S::display_id()
+                );
+                return;
+            }
+        }
+    }
+}
+
+async fn scrape_block_range<S>(
+    rpc_client: &EthRpcClient,
+    contract_address: Address,
+    topics: Vec<Topic>,
+    block_range: BlockRangeInclusive,
+) -> Result<(), MultiCallError<Vec<LogEntry>>>
+where
+    S: LogScraping,
+{
+    let mut subranges = VecDeque::new();
+    subranges.push_back(block_range);
+
+    while !subranges.is_empty() {
+        let range = subranges.pop_front().unwrap();
+        let (from_block, to_block) = range.clone().into_inner();
+
+        let request = GetLogsParam {
+            from_block: BlockSpec::from(from_block),
+            to_block: BlockSpec::from(to_block),
+            address: vec![contract_address],
+            topics: topics.clone(),
+        };
+
+        let result = rpc_client
+            .eth_get_logs(request)
+            .await
+            .map(<S::Parser>::parse_all_logs);
+
+        match result {
+            Ok((events, errors)) => {
+                register_deposit_events(S::display_id(), events, errors);
+                mutate_state(|s| S::update_last_scraped_block_number(s, to_block));
+            }
+            Err(e) => {
+                log!(
+                    INFO,
+                    "Failed to get {} logs in range {range}: {e:?}",
+                    S::display_id()
+                );
+                if e.has_http_outcall_error_matching(HttpOutcallError::is_response_too_large) {
```

Comment:
Nit: `is_response_too_large` sounds like a question. Shouldn't it be `response_is_too_large` (or just `response_too_large`)?

Reply:
I personally like the convention that boolean getters have a name prefixed by `is_`, this is make it clear just by reading the code what the signature of that method should look like so I don't mind changing the method name but I would keep that prefix.

Reply:
Closing this thread to merge. Feel free to ping me if you want to discuss it further.